### PR TITLE
Handle wNAM transfers differently from other ERC20 transfers from Ethereum

### PR DESCRIPTION
**Pull request by james-chf**
Relates to https://github.com/anoma/namada/issues/989 and https://github.com/anoma/namada/issues/508

This PR 
- makes it so that transfers of the wNAM ERC20 (i.e. transfers of ERC20s with an address that matches the `native_erc20` config value in chain storage) do not result in the minting of wrapped ERC20s on Namada. For now, we no-op, in a later PR, we will release NAM from escrow
- adds some functionality for testing the Ethereum bridge using the fake events endpoint in e2e tests, and stubs out a test for wNAM transfers (which right now just greps for the log line indicating that the wNAM transfer was received)
- moves a helper fn for reading the `native_erc20` address from storage from the bridge pool VP to the `namada_ethereum_bridge` crate, so it can be used more generally (e.g. as in this PR)

### CODE REVIEW

File: ethereum_bridge/src/parameters.rs [hunk 1]
```diff
@@ -218,6 +219,31 @@ impl EthereumBridgeConfig {
...
+    H: storage::traits::StorageHasher,
+{
+    let native_erc20 = bridge_storage::native_erc20_key();
+    match storage.read(&native_erc20) {
+        Ok((Some(bytes), _)) => {
+            Ok(EthAddress::try_from_slice(bytes.as_slice()).expect(
+                "Deserializing the Native ERC20 address from storage \
+                 shouldn't fail.",
+            ))
+        }
```
Comment (sug0) [inline ethereum_bridge/src/parameters.rs:237]: call `StorageRead::read` from `core` instead, maybe? it does a borsh deserialize automatically

File: ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs [hunk 2]
```diff
@@ -41,53 +46,76 @@ where
...
-        })?;
-        _ = changed_keys.insert(supply_key);
+        if asset != &wrapped_native_erc20 {
+            let mut changed =
+                mint_wrapped_erc20s(storage, asset, receiver, amount)?;
+            changed_keys.append(&mut changed)
+        } else {
+            tracing::warn!(
+                "Redemption of the wrapped native token is not yet supported"
+            )
```
Comment (sug0) [inline ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs:64]: maybe this is a bit silly, but we could log here the receiver and amount that would have been applied in storage

**Review by sug0 — CHANGES_REQUESTED** [review]

File: ethereum_bridge/src/parameters.rs (hunk 1 above)
Comment (james-chf) [reply ethereum_bridge/src/parameters.rs:237]: Yeah `StorageRead::read` is much nicer!

File: ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs (hunk 2 above)
Comment (james-chf) [reply ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs]: Sure, have updated the e2e test to check those also

**Review by sug0 — APPROVED** [review]

**Review by batconjurer — APPROVED** [review]

**Pull request merged**